netutil/netutil_test.go

```diff
@@ -0,0 +1,116 @@
+package netutil
+
+import (
+	"net"
+	"os"
+	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/stretchr/testify/assert"
+)
+
+// Setup required logger and example interface names and addresses
+var (
+	logger        log.Logger = log.NewLogfmtLogger(os.Stdout)
+	testIntsAddrs            = map[string]string{
```

Comment:
Can we remove this global var at all? I think all your unit tests can be written in a single function using the go table testing pattern.

Reply:
I also would like to see the following tests:
- An interface with multiple addressed
- Unable to parse the address (shouldn't happen, but we handle it in the code so let's test it)

Reply:
I got rid of the global variables and refactored the testing code using the go table testing pattern. I also added a test case for an address that cannot be parsed.